doc/memory_statistics/memory_statistics_hld.md

```diff
@@ -163,46 +163,69 @@ The high-level feature design diagram is shown below.
+
+3. **Enable Daemon Manually**: Administrators need to manually enable the `memorystatsd` process via the CLI before it starts collecting memory statistics.
+
+4. **Runtime Configuration Changes**: Administrators can modify settings like retention periods or sampling intervals via the CLI. These changes are written directly to the `MEMORY_STATISTICS` in ConfigDB.
+
+5. **Monitor ConfigDB for Changes**: `hostcfgd` continuously monitors ConfigDB for updates to the `MEMORY_STATISTICS`.
+
+6. **Signal Daemon to Reload Configuration**: Upon detecting changes in ConfigDB, `hostcfgd` sends a `SIGHUP` signal to the `memorystatsd` daemon, prompting it to reload its configuration without restarting.
+
+7. **Reload Daemon**: The `memorystatsd` process applies the new settings from ConfigDB dynamically, allowing the daemon to continue operating with updated parameters during runtime.
+
+8. **Handling Crashes and Restarts**: In case of a crash or restart, the `memorystatsd` daemon will always reload its default settings from the config file, ensuring a consistent startup state. Runtime-configured values from ConfigDB are only applied during runtime and not retained after a restart unless manually reloaded.
+
+9. **Graceful Shutdown**: When the daemon needs to be stopped, the `SIGTERM` signal ensures a graceful shutdown, where the daemon cleans up resources and terminates smoothly.
 
 ### Config DB Enhancements
 
-A new table, `MEMORY_STATISTICS_TABLE`, will be introduced in `ConfigDB` to store the configuration settings of the Memory Statistics feature. This table will allow for management of data collection frequency, retention period, and enable/disable status. The relevant configuration parameters and the schema for this table are detailed below.
+A new table, `MEMORY_STATISTICS`, will be introduced in `ConfigDB` to store the configuration settings of the Memory Statistics feature. This table will allow for management of data collection frequency, retention period, and enable/disable status. The relevant configuration parameters and the schema for this table are detailed below.
 
 **MEMORY_STATS Configuration Parameters**
 
 | Parameter           | Type        | Description                                                    |
 |---------------------|-------------|----------------------------------------------------------------|
 | enabled             | boolean     | Enable or disable memory statistics collection.                |
-| sampling_interval   | unit16      | Interval for memory data collection.       		     |
-| retention_period    | unit16      | Duration for which memory data is retained.		     |   
+| sampling_interval   | unit8      | Interval for memory data collection.       		     |
+| retention_period    | unit8      | Duration for which memory data is retained.		     |   
```

Comment:
What are the units for interval and period? in seconds? IF yes, is uint8 too small?
typo: unit8->uint8

Reply:
Thank you for your valuable feedback. To clarify:

- The **sampling_interval** is provided by the user in minutes, with a configurable range from **3 to 15** minutes.
- The **retention_period** is provided by the user in days, with a configurable range from **1 to 30** days.

Since the values for sampling_interval and retention_period fall within the range of **uint8** (0 to 255), we believe uint8 is appropriate as the data type for representing the **user input**.

Additionally, the typo has been corrected to uint8 as suggested and the descriptions have been updated for further clarity